llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -187,6 +196,8 @@ class SDWADstPreserveOperand : public SDWADstOperand {
         Preserve(PreserveOp) {}
 
   bool convertToSDWA(MachineInstr &MI, const SIInstrInfo *TII) override;
+  bool canCombineSelections(const MachineInstr &MI,
```

Comment:
Not sure this is needed

Reply:
It was strictly necessary as long as the check was outside of `potentialToConvert`; see also https://github.com/llvm/llvm-project/pull/123221#discussion_r1967148059.